flask_restless/helpers.py

```diff
@@ -488,6 +472,40 @@ def __call__(self, model, _apimanager=None, **kw):
         raise ValueError(message)
 
 
+class PrimaryKeyFinder(KnowsAPIManagers, Singleton):
+    """The singleton class that backs the :func:`primary_key_for` function."""
+
+    def __call__(self, model, _apimanager=None, **kw):
```

Comment:
This seems unnecessarily complicated. Let me think of a simpler way to do this...